crypto/engine/eng_devcrypto.c

```diff
@@ -340,6 +411,8 @@ static void prepare_cipher_methods(void)
                                           | EVP_CIPH_FLAG_DEFAULT_ASN1)
             || !EVP_CIPHER_meth_set_init(known_cipher_methods[i], cipher_init)
             || !EVP_CIPHER_meth_set_do_cipher(known_cipher_methods[i],
+                                     cipher_mode == EVP_CIPH_CTR_MODE ?
+                                              ctr_do_cipher :
                                               cipher_do_cipher)
```

Comment:
I'm pondering if `cipher_data` should have another field for the function to be called...  dunno.

Reply:
As long as we only have a couple, it may be fine as is.  If we were to add another one, I'd do it for sure.  Right now, this looks less ugly than having cipher_data defined after the functions, and easier than writing a proper header file (please call me lazy).

Reply:
I will call you lazy, and then tell you that I share that laziness :wink:

So ok, this is fine